tests/test_grpc_functions.py

```diff
@@ -0,0 +1,248 @@
+        port = find_free_port()
+        env = {
+            'PYTHONPATH': '%s/tests/functions:$PYTHONPATH' % os.getcwd(),
+             'GRPC_PORT': str(port),
+            'FUNCTION_URI': 'file://%s/tests/functions/concat.py?handler=concat' % os.getcwd()
+        }
+
+        self.process = subprocess.Popen(self.command,
+                                        cwd=self.workingdir,
+                                        shell=True,
+                                        env=env,
+                                        preexec_fn=os.setsid,
+                                        )
+        time.sleep(0.5)
+
+        channel = grpc.insecure_channel('localhost:%d' % port)
+        self.stub = function.MessageFunctionStub(channel)
+
+        def generate_messages():
+            headers = {
+                'Content-Type': message.Message.HeaderValue(values=['application/json']),
+                'Accept': message.Message.HeaderValue(values=['text/plain']),
+                'correlationId': message.Message.HeaderValue(values=[str(uuid.uuid4())])
+            }
+
+            messages = [
+                message.Message(payload='{"foo":"bar","hello":"world"}', headers=headers),
+            ]
+            for msg in messages:
+                yield msg
+
+        responses = self.stub.Call(generate_messages())
+
+        for response in responses:
+            self.assertEquals('{"result": "foobarhelloworld"}', response.payload)
+
+    def test_accepts_not_supported(self):
+        port = find_free_port()
+        env = {
+            'PYTHONPATH': '%s/tests/functions:$PYTHONPATH' % os.getcwd(),
+            'GRPC_PORT': str(port),
+            'FUNCTION_URI': 'file://%s/tests/functions/concat.py?handler=concat' % os.getcwd()
+        }
+
+        self.process = subprocess.Popen(self.command,
+                                        cwd=self.workingdir,
+                                        shell=True,
+                                        env=env,
+                                        preexec_fn=os.setsid,
+                                        )
+        time.sleep(0.5)
+
+        channel = grpc.insecure_channel('localhost:%s' % port)
+        self.stub = function.MessageFunctionStub(channel)
+
+        def generate_messages():
+            headers = {
+                'Content-Type': message.Message.HeaderValue(values=['application/json']),
+                'Accept': message.Message.HeaderValue(values=['application/xml']),
+                'correlationId': message.Message.HeaderValue(values=[str(uuid.uuid4())])
+            }
+
+            messages = [
+                message.Message(payload='{"foo":"bar","hello":"world"}', headers=headers),
+            ]
+            for msg in messages:
+                yield msg
+
+        try:
+            responses = self.stub.Call(generate_messages())
+            self.assertEquals(grpc._channel._Rendezvous, type(responses))
+            #TODO: Investigate error handling
```

Comment:
Did you forget to address this TODO? If not, it would be worth logging an issue on github and noting the issue in the TODO.

Reply:
Didn't forget, but will log an issue